src/components/select-menu/useSelectMenuAnimations.tsx

```diff
@@ -0,0 +1,199 @@
+
+  if (placement.includes('top')) {
+    transformTopAmount = 8;
+  }
+
+  if (!originalElementRef.current) return;
+
+  floatingContainerElement.style.top = `${
+    originalElementRef.current.top + transformTopAmount
+  }px`;
+};
+
+const useSelectMenuAnimations = (props: UseSelectMenuAnimationsPropsType) => {
+  const {
+    selectId,
+    popupClassName,
+    popupContentClassName,
+    floatingContainerClassName,
+    selectElementClassName,
+  } = props;
+  const lastRef = React.useRef<DOMRect>();
+  const selectRef = React.useRef<DOMRect>();
+  const hasReduceMotion = useReducedMotion();
+
+  const animateExit = ({callback}) => {
+    const select = document.getElementById(selectId);
+    const popupContainer = select.getElementsByClassName(
+      popupClassName
+    )[0] as HTMLDivElement;
+    const floatingContainer = select.getElementsByClassName(
+      floatingContainerClassName
+    )[0] as HTMLDivElement;
+    const popupContent = select.getElementsByClassName(
+      popupContentClassName
+    )[0] as HTMLDivElement;
+
+    popupContainer.classList.add('exit-animation');
+    popupContainer.style.height = `0px`;
+    if (selectRef.current)
+      popupContainer.style.width = `${selectRef.current.width}px`;
+    popupContainer.style.opacity = `0`;
+    popupContent.classList.add(SCROLL_HIDE_CLASSNAME);
+
+    requestAnimationFrame(() => {
+      popupContainer.classList.add('animate-on-transforms');
+      floatingContainer.classList.add('animate-on-transforms');
+      resetFloatingContainerTopPosition(floatingContainer, lastRef);
+
+      if (callback) callback();
+    });
+  };
+
+  const animateEntry = () => {
+    // Wait for the next frame
+    // so we allow the floating container to apply all flip styles
+    requestAnimationFrame(() => {
+      const select = document.getElementById(selectId);
+
+      if (!select) return;
+
+      const floatingContainer = select.getElementsByClassName(
+        floatingContainerClassName
+      )[0] as HTMLDivElement;
+      const popupContainer = select.getElementsByClassName(
+        popupClassName
+      )[0] as HTMLDivElement;
+      const popupContent = select.getElementsByClassName(
+        popupContentClassName
+      )[0] as HTMLDivElement;
+      const selectElement = select.getElementsByClassName(
+        selectElementClassName
+      )[0] as HTMLDivElement;
+
+      // Register desired position
+      lastRef.current = floatingContainer.getBoundingClientRect();
+      selectRef.current = selectElement.getBoundingClientRect();
+      const initialContainerSize = lastRef.current;
+      const selectElementSize = selectRef.current;
+
+      popupContent.style.width = `${initialContainerSize.width}px`;
+      popupContent.style.height = `${initialContainerSize.height}px`;
+      popupContent.classList.add(SCROLL_HIDE_CLASSNAME);
+
+      if (!hasReduceMotion) {
+        // Reset the popup height to the pre-appear position
+        popupContainer.style.height = `1px`;
```

Comment:
why 1 px?

Reply:
hmm I had some issues with the animation in the past when it was set to 0, but it seems it works all fine now with 0! Thanks for catching this